### DISCUSSION TITLE
MonoGo - Cross-Platform .NET 8 C# 2D Game Engine

### DISCUSSION THREAD

**Discussion by BlizzCrafter** [MonoGame/MonoGame#8504] [Showcase]
![Banner](https://raw.githubusercontent.com/BlizzCrafter/MonoGo/refs/heads/master/logos/Banner.png)

# Welcome to MonoGo!
[![Version](https://img.shields.io/nuget/v/MonoGo.Engine?style=for-the-badge&logo=git&logoColor=E73C00&logoSize=auto&label=MonoGo.Engine&labelColor=262626&color=707070)](https://github.com/MonoGo-Engine) [![NuGet](https://img.shields.io/badge/NuGet-MonoGo.Templates-blue.svg?style=for-the-badge&logo=NuGet&logoColor=0289CC&logoSize=auto&colorA=262626&colorB=707070)](https://www.nuget.org/packages/MonoGo.Templates) [![Docs](https://img.shields.io/github/labels/MonoGo-Engine/MonoGo/Documentation?style=for-the-badge&logo=gitbook&logoColor=E2B004&logoSize=auto&labelColor=262626&color=707070)](https://monogo-engine.github.io/monogo.github.io/)

Cross-Platform .NET 8 C# 2D game engine build ontop of MonoGame.

# Features
* Graphics Pipeline and Automated Batch\Vertex Buffer Management.
* Easy SpriteSheet-Animations.
* Texture Packing.
* Sprite Groups and Dynamic Graphics Loading.
* Input Management with GamePad support.
* Timers, Alarms, Cameras, State Machines.
* Coroutines.
* Hybrid EC.
* Scene System with Layers.
* GameDev related Math lib.
* Lightweight Collision Detection.
* Integrated Post-Processing Management (Includes Bloom & ColorGrading Effects).
* Powerful Particle-Effect-System.
* Extensive GUI-System.
* Enhanced Content Management.
* FMOD Audio Support (As a standalone [Library](https://github.com/Martenfur/FmodForFoxes/)).
### Modules:
* Tiled Map Support via [Tiled](https://www.mapeditor.org/).
* Iguina UI-System Support via [Iguina](https://github.com/RonenNess/Iguina).

# Sample Project

The **[sample project](https://github.com/MonoGo-Engine/MonoGo.Samples)** itself contains alot more demos, but here are some impressions at least:

![Color Picker](https://raw.githubusercontent.com/MonoGo-Engine/MonoGo.Samples/refs/heads/master/docs/sample_00.jpg)
![Tiled](https://raw.githubusercontent.com/MonoGo-Engine/MonoGo.Samples/refs/heads/master/docs/sample_01.jpg)
![PostFX](https://raw.githubusercontent.com/MonoGo-Engine/MonoGo.Samples/refs/heads/master/docs/sample_02.jpg)

Click **[here](https://github.com/MonoGo-Engine/MonoGo.Samples)** to see more engine features!

# Now Have Fun with MonoGo!

![Banner](https://raw.githubusercontent.com/BlizzCrafter/MonoGo/refs/heads/master/logos/Social.png)

# Now on GitHub:

**Comment by nightblade9** [comment]
Looks very cool, thank you for sharing! Seems like there are quite a few editor-ish projects, this might be just the thing I need to bootstrap mine :)

**Comment by nightblade9** [comment]
Are there any tutorials for this? I see a samples repo, I see an API doc, but no actual "here's how you start" kinda tutorial thing. I'm particularly interested in the UI right now.
